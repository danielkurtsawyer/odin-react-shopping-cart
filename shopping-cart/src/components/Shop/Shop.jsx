import styles from "./shop.module.css";
import ItemCard from "../ItemCard/ItemCard";
import PropTypes from "prop-types";

function Shop({ products, addToCart }) {
  console.log(products);
  return (
    <div className={styles.shopContainer} data-testid="shop-container">
      {products &&
        products.map((product) => {
          return (
            <ItemCard
              key={product.id}
              id={product.id}
              title={product.title}
              price={product.price}
              image={product.image}
              addToCart={addToCart}
            />
          );
        })}
    </div>
  );
}

Shop.propTypes = {
  products: PropTypes.object.isRequired,
  addToCart: PropTypes.func.isRequired,
};

export default Shop;
