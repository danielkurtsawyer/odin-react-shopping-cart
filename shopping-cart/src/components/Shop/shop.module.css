.shopContainer {
  padding: 60px;
}
