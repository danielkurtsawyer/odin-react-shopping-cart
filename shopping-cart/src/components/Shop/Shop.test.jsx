import { vi, describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";

import Shop from "./Shop";

describe("Shop", () => {
  it("should display each product and render correctly via snapshot", () => {
    const addToCart = vi.fn();
    const products = [
      {
        id: 1,
        title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
        price: 109.95,
        description:
          "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
        category: "men's clothing",
        image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_t.png",
        rating: {
          rate: 3.9,
          count: 120,
        },
      },
      {
        id: 2,
        title: "Mens Casual Premium Slim Fit T-Shirts ",
        price: 22.3,
        description:
          "Slim-fitting style, contrast raglan long sleeve, three-button henley placket, light weight & soft fabric for breathable and comfortable wearing. And Solid stitched shirts with round neck made for durability and a great fit for casual fashion wear and diehard baseball fans. The Henley style round neckline includes a three-button placket.",
        category: "men's clothing",
        image:
          "https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_t.png",
        rating: {
          rate: 4.1,
          count: 259,
        },
      },
    ];
    const shopComponent = render(
      <Shop products={products} addToCart={addToCart} />
    );

    expect(shopComponent).toMatchSnapshot();
  });

  it("should not render Item Cards if products array is null or not provided", () => {
    render(<Shop products={null} />);

    const shopContainer = screen.getByTestId("shop-container");
    expect(shopContainer.children.length).toEqual(0);
  });
});
