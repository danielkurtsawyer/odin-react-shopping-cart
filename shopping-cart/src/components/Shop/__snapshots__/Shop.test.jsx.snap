// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop > should display each product and render correctly via snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_shopContainer_3c5ce2"
        data-testid="shop-container"
      >
        <div
          class="_itemCardContainer_6a0a13"
        >
          <img
            alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
            class="_itemCardImage_6a0a13"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_t.png"
          />
          <div
            class="_itemCardTitle_6a0a13"
          >
            Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
          </div>
          <div
            class="_itemCardPrice_6a0a13"
          >
            $
            109.95
          </div>
          <div
            class="_itemCardQuantity_6a0a13"
          >
            <img
              alt="-"
              src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M200-440v-80h560v80H200Z'/%3e%3c/svg%3e"
            />
            <input
              aria-valuenow="1"
              min="1"
              name="quantity"
              type="number"
              value="1"
            />
            <img
              alt="+"
              src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M440-440H200v-80h240v-240h80v240h240v80H520v240h-80v-240Z'/%3e%3c/svg%3e"
            />
          </div>
          <button>
            Add to Cart
          </button>
        </div>
        <div
          class="_itemCardContainer_6a0a13"
        >
          <img
            alt="Mens Casual Premium Slim Fit T-Shirts "
            class="_itemCardImage_6a0a13"
            src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_t.png"
          />
          <div
            class="_itemCardTitle_6a0a13"
          >
            Mens Casual Premium Slim Fit T-Shirts 
          </div>
          <div
            class="_itemCardPrice_6a0a13"
          >
            $
            22.30
          </div>
          <div
            class="_itemCardQuantity_6a0a13"
          >
            <img
              alt="-"
              src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M200-440v-80h560v80H200Z'/%3e%3c/svg%3e"
            />
            <input
              aria-valuenow="1"
              min="1"
              name="quantity"
              type="number"
              value="1"
            />
            <img
              alt="+"
              src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M440-440H200v-80h240v-240h80v240h240v80H520v240h-80v-240Z'/%3e%3c/svg%3e"
            />
          </div>
          <button>
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_shopContainer_3c5ce2"
      data-testid="shop-container"
    >
      <div
        class="_itemCardContainer_6a0a13"
      >
        <img
          alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
          class="_itemCardImage_6a0a13"
          src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_t.png"
        />
        <div
          class="_itemCardTitle_6a0a13"
        >
          Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
        </div>
        <div
          class="_itemCardPrice_6a0a13"
        >
          $
          109.95
        </div>
        <div
          class="_itemCardQuantity_6a0a13"
        >
          <img
            alt="-"
            src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M200-440v-80h560v80H200Z'/%3e%3c/svg%3e"
          />
          <input
            aria-valuenow="1"
            min="1"
            name="quantity"
            type="number"
            value="1"
          />
          <img
            alt="+"
            src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M440-440H200v-80h240v-240h80v240h240v80H520v240h-80v-240Z'/%3e%3c/svg%3e"
          />
        </div>
        <button>
          Add to Cart
        </button>
      </div>
      <div
        class="_itemCardContainer_6a0a13"
      >
        <img
          alt="Mens Casual Premium Slim Fit T-Shirts "
          class="_itemCardImage_6a0a13"
          src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_t.png"
        />
        <div
          class="_itemCardTitle_6a0a13"
        >
          Mens Casual Premium Slim Fit T-Shirts 
        </div>
        <div
          class="_itemCardPrice_6a0a13"
        >
          $
          22.30
        </div>
        <div
          class="_itemCardQuantity_6a0a13"
        >
          <img
            alt="-"
            src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M200-440v-80h560v80H200Z'/%3e%3c/svg%3e"
          />
          <input
            aria-valuenow="1"
            min="1"
            name="quantity"
            type="number"
            value="1"
          />
          <img
            alt="+"
            src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%231f1f1f'%3e%3cpath%20d='M440-440H200v-80h240v-240h80v240h240v80H520v240h-80v-240Z'/%3e%3c/svg%3e"
          />
        </div>
        <button>
          Add to Cart
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
